File: modules/agent/agents/explore.go
package agents

import (
	"github.com/woyin/OrangeCoding/modules/ai"
	"github.com/woyin/OrangeCoding/modules/core"
	"github.com/woyin/OrangeCoding/modules/tools"
)

// NewExplore creates the codebase exploration agent.
// RoleObserver, tools: [read_file, find, grep, glob].
func NewExplore(provider ai.AiProvider, registry *tools.ToolRegistry, workDir string) *BaseAgent {
	allowedTools := []string{"read_file", "find", "grep", "glob"}
	return newFilteredAgent(provider, registry, workDir, core.RoleObserver, allowedTools,
		"You are Explorer, the codebase navigator. You explore codebases, map structure, identify patterns, and report findings clearly.")
}

File: packages/agent/src/agents/explore.ts
/**
 * Explore - codebase exploration agent.
 * Ported from modules/agent/agents/explore.go.
 */

import type { AiProvider } from "@orangecoding/ai";
import { AgentRole } from "@orangecoding/core";
import { ToolRegistry } from "@orangecoding/tools";
import { BaseAgent } from "./base.js";
import { newFilteredAgent } from "./helpers.js";

export function newExplore(provider: AiProvider, registry: ToolRegistry, workDir: string): BaseAgent {
  return newFilteredAgent(provider, registry, workDir, AgentRole.Observer,
    ["read_file", "find", "grep", "glob"],
    "You are Explorer, the codebase navigator. You explore codebases, map structure, identify patterns, and report findings clearly.");
}
